repos/fdbt-site/src/pages/api/defineTimeRestrictions.ts

```diff
@@ -125,6 +156,25 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
             return;
         }
 
+        // if in edit mode
+        if (ticket && matchingJsonMetaData) {
```

Comment:
Do we need this code twice? Can we not just have it once, to check if we're in edit mode and set timeRestriction to undefined if they didn't select one, and a time restriction if they did set one?

Reply:
we cannot as if we have to do that, then we need to add the error handling and other stuff in the edit mode and will make the code complex.

Reply:
I think we need to refactor this API. If you look in the page (repos/fdbt-site/src/pages/chooseTimeRestrictions.tsx) and look for the inputs with name="timeRestrictionChoice", you can see there is only 2 choices - Premade and No. The API has a load of code incase the user selects 'Yes' - which isnt an option. We need to delete that code, and then we can just have our editMode code once.

Reply:
I think a lot of tests can also probably be deleted once we do this, because they will be checking what happens when the user inputs 'Yes' via the API, but thats not possible.

Reply:
okiess... even i got confused with the option of "Yes" ... let me refactor the code